### PR TITLE
Validate initializers before writing to memory

### PR DESCRIPTION
**Pull request by bjfish**
Fixes `linking.wast:342`

### CODE REVIEW

**Comment by bjfish** [comment]
bors try

File: lib/runtime-core/src/backing.rs [hunk 1]
```diff
@@ -143,17 +145,49 @@ impl LocalBacking {
...
                     if memory_desc.minimum.bytes().0 < data_top || data_top < init_base {
                         return Err(vec![LinkError::Generic {
                             message: "data segment does not fit".to_string(),
                         }]);
                     }
+                }
+                LocalOrImport::Import(imported_memory_index) => {
+                    // Write the initialization data to the memory that
+                    // we think the imported memory is.
+                    unsafe {
```
Comment (syrusakbary) [inline lib/runtime-core/src/backing.rs]: Can we move the unsafe only for the local_memory retrieval?

```rust
let local_memory = unsafe { &*imports.vm_memories[imported_memory_index] };
 ```

Comment (bjfish) [reply lib/runtime-core/src/backing.rs]: Done, I also minimized the other `unsafe` block at 38a8a0eb01d2802f7a5899e7276653d075f21a2d

**Pull request merged**